File: engine/tests/fixtures/page_objects/register.page.ts
import type { Locator, Page } from "@playwright/test";

export class RegisterPage {
  readonly emailInput: Locator;
  readonly passwordInput = this.page.locator('input[type="password"][name="password"][minlength="12"][maxlength="72"]');
  readonly mobilePhoneField = this.page.getByPlaceholder("Mobile phone");
  readonly spendLimitInput = this.page.locator('input[type="number"][name="spendLimit"][min="0"][max="500"]');
  readonly submitButton = this.page.getByRole("button", { name: "Create account" });

  constructor(private readonly page: Page) {
    this.emailInput = page.getByLabel("Work email");
  }

  async fillWebsite(website: string) {
    await this.page.fill('input[type="url"][name="website"]', website);
  }
}

File: engine/tests/fixtures/page_objects/RegisterPage.java
package fixtures.pageobjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class RegisterPage {
    private final WebDriver driver;

    @FindBy(css = "input[type='email'][name='email'][required]")
    private WebElement emailInput;

    private final By passwordInput = By.cssSelector("input[type='password'][name='password'][minlength='12'][maxlength='72']");
    private final By mobilePhoneField = By.name("mobilePhone");
    private final By spendLimitInput = By.cssSelector("input[type='number'][name='spendLimit'][min='0'][max='500']");

    public RegisterPage(WebDriver driver) {
        this.driver = driver;
    }

    public void fillWebsite(String website) {
        driver.findElement(By.cssSelector("input[type='url'][name='website']")).sendKeys(website);
    }

    public void submit() {
        driver.findElement(By.cssSelector("button[type='submit']")).click();
    }
}
